packages/components/src/components/toolbar/shadow.tsx

```diff
@@ -0,0 +1,159 @@
+import type { JSX } from '@stencil/core';
+import { Component, Element, h, Host, Listen, Prop, State, Watch } from '@stencil/core';
+
+import type { LabelPropType, ToolbarAPI, ToolbarStates, ToolbarItemsPropType, ToolbarItemPropType } from '../../schema';
+import { validateLabel, validateToolbarItems } from '../../schema';
+import { KolLinkTag, KolButtonTag } from '../../core/component-names';
+
+const TOOLBAR_ITEM_TAG_NAME = 'kol-toolbar-item';
+
+@Component({
+	tag: 'kol-toolbar',
+	styleUrls: {
+		default: './style.scss',
+	},
+	shadow: true,
+})
+export class KolToolbar implements ToolbarAPI {
+	@Element() host!: HTMLElement;
+
+	@State() public state: ToolbarStates = {
+		_label: '',
+		_items: [],
+	};
+
+	@State() private currentIndex: number = 0;
+
+	private renderElement = (element: ToolbarItemPropType, index: number): JSX.Element => {
+		const tabIndex = index === this.currentIndex && !element?._disabled ? 0 : -1;
+		const props = {
+			key: index,
+			class: TOOLBAR_ITEM_TAG_NAME,
+			_tabIndex: tabIndex,
+		};
+		if ('_href' in element) {
+			return <KolLinkTag {...element} {...props}></KolLinkTag>;
+		}
+		return <KolButtonTag {...element} {...props}></KolButtonTag>;
+	};
+
+	public render(): JSX.Element {
+		return (
+			<Host class="kol-toolbar">
+				<div class="toolbar" role="toolbar" aria-label={this.state._label}>
+					{(this.state._items as ToolbarItemPropType[]).map(this.renderElement)}
+				</div>
+			</Host>
+		);
+	}
+
+	/**
+	 * Defines the semantic aria-label of the component.
+	 */
+	@Prop() public _label!: string;
+
+	/**
+	 * Defines the functional elements of toolbar to render (e.g. kol-link, kol-button).
+	 */
+	@Prop() public _items!: ToolbarItemsPropType;
+
+	@Watch('_label') validateLabel(value?: LabelPropType): void {
+		validateLabel(this, value);
+	}
+
+	@Watch('_items')
+	public validateItems(value?: ToolbarItemsPropType): void {
+		validateToolbarItems(this, value);
+	}
+
+	/**
+	 * Retrieves the toolbar.
+	 *
+	 * @returns An array of HTMLElements representing the toolbar items.
+	 */
+	private getToolbar(): HTMLElement | null | undefined {
+		return this.host.shadowRoot?.querySelector('.toolbar');
```

Comment:
Ich würde gerne alle DOM-Queries durch Refs ablösen, da diese nativ von Stencil unterstützt werden: https://stenciljs.com/docs/templating-jsx#getting-a-reference-to-a-dom-element

Betrifft `getToolbar` (die Methode wird dann überflüssig) und `getToolbarItems`